airbyte-workers/src/main/java/io/airbyte/workers/temporal/SyncWorkflow.java

```diff
@@ -141,21 +151,62 @@ public StandardSyncOutput replicate(JobRunConfig jobRunConfig,
         return syncInput;
       };
 
-      final TemporalAttemptExecution<StandardSyncInput, StandardSyncOutput> temporalAttemptExecution = new TemporalAttemptExecution<>(
-          workspaceRoot,
-          jobRunConfig,
-          getWorkerFactory(sourceLauncherConfig, destinationLauncherConfig, jobRunConfig, syncInput),
-          inputSupplier,
-          new CancellationHandler.TemporalCancellationHandler());
+      final Predicate<ReplicationOutput> shouldAttemptAgain =
+          output -> output.getReplicationAttemptSummary().getStatus() != ReplicationStatus.COMPLETED;
```

Comment:
to make sure I get it right - this check is necessary because we can have state even though the job did not correctly finish?

Reply:
exactly. the idea is that even on a failure if we got a state from the destination we trust that the destination committed data up until that point, so that is where we should start from on the next sync.

Reply:
makes sense. thanks for explaining.